effects/effect.py

```diff
@@ -17,8 +15,75 @@ class EffectStatus(Enum):
 
 class Effect(ABC):
 
-    def __init__(self, name: str, game: Game, creator: Citizen) -> None:
-        self.name = name
-        self.game = game
-        self.targets: List[Citizen] = []
-        self.creator: Citizen = creator
+    def __init__(self,
+                 game: Game,
+                 name: str,
+                 creator: Citizen,
+                 priority: int = 0) -> None:
+
+        self._game = game
+        self._name = name
+        self._creator = creator
+        self._PRIORITY = priority
+
+        self._targets: List[Citizen] = []
+        self._status = EffectStatus.CREATED
+        self._activation_round = 0
+
+    def activate(self) -> None:
+        if self._status == EffectStatus.CREATED:
+            is_activated = self._activate_impl()
+
+            if (is_activated):
+                self._status = EffectStatus.ACTIVATED
+                self._activation_round = self.game.round_number
```

Comment:
Possibly, warning should be added. Effect should be always activated succesfully

Reply:
Done.